gateway/api/serializers.py

```diff
@@ -242,3 +245,106 @@ class RuntimeJobSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = RuntimeJob
+
+
+class CatalogProviderSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the Provider model in the Catalog View.
+    """
+
+    class Meta:
+        model = Provider
+        fields = ["name", "icon_url"]
+
+
+class ListCatalogSerializer(serializers.ModelSerializer):
+    """
+    List Serializer for the Catalog View.
+    """
+
+    provider = CatalogProviderSerializer()
+    available = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Program
+
+    def get_available(self, obj):
+        """
+        This method populates available field.
+        If the user has RUN PERMISSION in any of its groups
+        available field will be True. If not, will be False.
+        """
+        author = self.context.get("author", None)
```

Comment:
nit

```suggestion
        user = self.context.get("author", None)
```

to avoid any confusion about intent (i.e. is the scope all functions the user has access to or only the functions the user is the author of)

Reply:
I see, I like this distinction between user of the request and author of the function.

Reply:
Applied change in [1c05730](https://github.com/Qiskit/qiskit-serverless/pull/1460/commits/1c057301ec0c23f735f9bf04ca9ef66ca2696851)

Let me know if this is what you were expecting, Paul.